File: GFG POTD/020-July-2025/024-last-moment-before-all-ants-fall-out-of-a-plank.java
// https://www.geeksforgeeks.org/problems/last-moment-before-all-ants-fall-out-of-a-plank/1




class Solution {
    // TC : O(N + M) Where N = length of right and M = length of left
    // SC : O(1)
    public int getLastMoment(int n, int left[], int right[]) {
        int lastTime = Integer.MIN_VALUE;
        for (int i = 0; i < right.length; i ++) {
            lastTime = Math.max(lastTime, n - right[i]);
        }
        for (int i = 0; i < left.length; i ++) {
            lastTime = Math.max(lastTime, left[i]);
        }
        return lastTime;
    }
}

File: GFG POTD/020-July-2025/024-last-moment-before-all-ants-fall-out-of-a-plank.cpp
// https://www.geeksforgeeks.org/problems/last-moment-before-all-ants-fall-out-of-a-plank/1




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N + M) where N = left.size() and M = right.size()
    // SC : O(1)
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        int lastTime = INT_MIN;
        for (int i = 0; i < right.size(); i ++) {
            lastTime = max(lastTime, n - right[i]);
        }
        for (int i = 0; i < left.size(); i ++) {
            lastTime = max(lastTime, left[i]);
        }
        return lastTime;
    }
};